pkg/templates/kubeadm/v1beta2/kubeadm.go

```diff
@@ -225,3 +225,53 @@ func NewConfig(s *state.State, host kubeoneapi.HostConfig) ([]runtime.Object, er
 
 	return []runtime.Object{initConfig, joinConfig, clusterConfig}, nil
 }
+
+// NewConfig returns all required configs to init a cluster via a set of v1beta2 configs
+func NewConfigWorker(s *state.State, host kubeoneapi.HostConfig) ([]runtime.Object, error) {
+	cluster := s.Cluster
+	nodeIP := host.PrivateAddress
+	if nodeIP == "" {
+		nodeIP = host.PublicAddress
+	}
+
+	nodeRegistration := kubeadmv1beta2.NodeRegistrationOptions{
+		Name: host.Hostname,
+		KubeletExtraArgs: map[string]string{
+			"anonymous-auth":      "false",
+			"node-ip":             nodeIP,
+			"read-only-port":      "0",
+			"rotate-certificates": "true",
```

Comment:
I am not sure. I don't have enough information about the scope.
@kron4eg What do you think?

Reply:
@xmudrii while all you've wrote is true, we have just the same setup for control-plane nodes.

Reply:
I've checked a couple of things since posting this comment.

> This feature requires the appropriate feature gates to be enabled. Do we enable those somewhere?

The needed feature gates/flags are now enabled by default, so there's nothing to do here.

> In case of the kubelet serving certificates, the CSRs can't be approved automatically. The approval process must be done by an operator or by a custom controller (we have such a controller in machine-controller, but we probably can't use it here). If the certificate is not approved, the kubelet will start misbehaving.

The `--rotate-certificates` flag is used to rotate only the client certificates, so we don't need to care about server certificates and NodeCSRApprover for now.

However, I have one more question/concern. Are the needed RBAC rules deployed on the cluster in the case when the machine-controller is not deployed? The needed RBAC rules can be found in the [Kubelet TLS bootstrapping](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping) document. I know that we deploy some similar rules when deploying machine-controller, but I'm not sure are those relevant just for the machine-controller.

Reply:
It seems like all needed rules are deployed by default.